lib/b_trigger.ml

```diff
@@ -840,42 +793,41 @@ let mouse_pos () =
 (* check if mouse didn't move for a while *)
 (* TODO use get_touch_finger *)
 let check_mouse_rest =
-  let pos0 = ref (0,0)
-  and t = ref (Some 0.) in
+  let t = ref None in
+  let on_mouse_idle () =
+    push_event @@ create_event mouse_at_rest
+  in
+  let start_timer () =
+      t := Some (mouse_pos (), Timeout.add 1000 on_mouse_idle)
+  in
   fun () ->
     match !t with
-    | None -> (* we start timer *)
-      t := Some (Unix.gettimeofday ());
-      pos0 := mouse_pos ();
-      0.
-    | Some t0 ->
+    | None -> start_timer ()
+    | Some (pos0, timeout)  ->
       let p = mouse_pos () in
-      if p <> !pos0 (* we have moved *)
-      then t := None;
-      Unix.gettimeofday () -. t0
+      if p <> pos0 (* we have moved *)
+      then begin
+        Timeout.cancel timeout;
+        start_timer ()
+      end
+
+let no_timeout () = -1
+
+let poll_noevent_fps = B_time.make_fps ()
```

Comment:
this function should be renewed at each run, otherwise the `start` hidden variable becomes meaningless